### ISSUE TITLE
Refactor BLS module

### ISSUE THREAD

**Issue by samlaf** [Layr-Labs/eigensdk-go#5]
We originally took much of the [crypto/bls](https://github.com/Layr-Labs/eigensdk-go/tree/master/crypto/bls) module from eigenDA, so it currently contains code for both bls and some incredible-squaring specific code.

We need to refactor it and take out all the incredible-squaring code and put that in its own repo.

**Comment by samlaf** [comment]
I really don't like that our bn254 and bls methods all [take pointer receivers and arguments](https://github.com/Layr-Labs/eigensdk-go/blob/5afbe24bcef6a5ad23722c0b1e5e139cd24083bf/crypto/bls/attestation.go#L94), eg:
```
func (p *G2Point) Add(p2 *G2Point)
```

I've been bitten a few times by side effects while writing the bls aggregation service. I think we just followed the gnark-crypto library design. I raised an [issue](https://github.com/Consensys/gnark-crypto/issues/461) in their remote asking for the rationale behind this design choice. In the meantime, I think we might want to switch to non pointer receivers and args. A few stack copies is not worth optimizing for, especially if it can save a bunch of human debugging time.
